src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -11306,6 +11306,43 @@ instruct convD2L_reg_reg(rRegL dst, regD src, rFlagsReg cr)
   ins_pipe(pipe_slow);
 %}
 
+instruct convF2HF_reg_reg(rRegI dst, regF src, regF tmp) %{
```

Comment:
You can move these patterns to common .ad file it will also handle 32 bit target.